### PR TITLE
sd-rtnl: make joining broadcast groups implicit

### PR DESCRIPTION
**Pull request by teg**

### CODE REVIEW

File: src/libsystemd/sd-rtnl/sd-rtnl.c [hunk 1]
```diff
@@ -1006,13 +984,39 @@ int sd_rtnl_add_match(sd_rtnl *rtnl,
...
 
         assert_return(rtnl, -EINVAL);
         assert_return(callback, -EINVAL);
         assert_return(!rtnl_pid_changed(rtnl), -ECHILD);
-        assert_return(rtnl_message_type_is_link(type) ||
-                      rtnl_message_type_is_addr(type) ||
-                      rtnl_message_type_is_route(type), -EOPNOTSUPP);
+
+        /* TODO: support all broadcast groups, and also leave them again when we no
+                 longer care */
```
Comment (dvdhrm) [inline src/libsystemd/sd-rtnl/sd-rtnl.c:994]: How about a simple ref-count per type? This way you'll leave the group again properly. And we can drop the TODO item then.

Comment (dvdhrm) [inline src/libsystemd/sd-rtnl/sd-rtnl.c:994]: per "type group", obviously. Not per-type.

Comment (teg) [inline src/libsystemd/sd-rtnl/sd-rtnl.c:994]: As discussed in person, this is not so trivial and we better figure out how genetlink will look first. I'll push with updated TODO.

**Comment by teg** [comment]
This was superseded by #171.

**Pull request closed without merging**